components/profile/pages/payments.tsx

```diff
@@ -29,185 +33,171 @@ export interface TotalFiatNetworks {
+
+  function TotalReceived() {
+    if (hasNoConvertedToken)
+      return(
+        <span className="caption-small text-danger">
+          {t("currencies.error-convert-all-to-euro")}
+        </span>
+      );
+
+    return(
+      <>
+        <span className="caption-medium font-weight-normal text-capitalize text-white mr-2">
+          {t("labels.recivedintotal")}
+        </span>
+
+        <div className="caption-large font-weight-medium bg-gray-900 py-2 px-3 border border-gray-850 border-radius-4">
+          <span className="text-white">
+            {formatNumberToCurrency(totalFiat)}
+          </span>
 
-  const {state} = useAppState();
-
-  const { getPayments } = useApi();
-
-
-  const [option, setOption] = useState<{ value: string; label: string }>(defaultOptions[0]);
-  const [startDate, setStartDate] = useState<string>(format(subDays(new Date(), 7), "yyyy-MM-dd").toString());
-  const [endDate, setEndDate] = useState<string>(format(new Date(), "yyyy-MM-dd").toString());
-
-  function onChangeSelect(e: { value: string; label: string }) {
-    setStartDate(e.value);
-    setEndDate(format(new Date(), "yyyy-MM-dd").toString());
-    setOption({
-      value: e.value,
-      label: e.label,
-    });
+          <span className="text-gray-600 ml-1">{state?.Settings?.currency?.defaultFiat}</span>
+        </div>
+      </>
+    );
   }
 
   useEffect(() => {
-    if (!state.currentUser?.walletAddress) return;
-
-    getPayments(state.currentUser.walletAddress, startDate, endDate).then(setPayments);
-  }, [state.currentUser?.walletAddress, startDate, endDate]);
+    if (!payments?.length) return;
 
-  useEffect(() => {
-    if (!payments) return;
-    const allNetworks: Network[] = [];
-    payments.map((payment) => {
-      if (
-        !networks.find((network) => network?.id === payment?.issue?.network?.id) &&
-        !allNetworks.find((network) => network?.id === payment?.issue?.network?.id)
-      ) {
-        allNetworks.push(payment?.issue?.network);
-        setNetworks(allNetworks);
-      }
-    });
+    Promise.all(payments.flatMap(({ payments }) => payments.map(async (payment) => ({
+      tokenAddress: payment?.issue?.transactionalToken?.address,
+      value: payment.ammount,
+      price: await getCoinPrice(payment?.issue?.transactionalToken?.symbol, state?.Settings?.currency?.defaultFiat),
+      networkId: payment?.issue?.network_id
+    }))))
+      .then((tokens) => {
+        const totalConverted = tokens.reduce((acc, token) => acc + token.value * (token.price || 0), 0);
+        const noConverted = !!tokens.find((token) => token.price === undefined);
+        
+        setTotalFiatNetworks(tokens)
+        setTotalFiat(totalConverted);
+        setHasNoConvertedToken(noConverted);
+      });
   }, [payments]);
 
   useEffect(() => {
-    if (!payments?.length) return;
+    setValue({ wallet: state.currentUser?.walletAddress || "" }, true);
+  }, [state.currentUser?.walletAddress]);
```

Comment:
take note that this effect does not trigger more than you want, as `state.` usage in useEffects should be avoided